Fixed bug for bus fault acknowledgment

AlessandroA opened the pull request:
Acknowledging the bus fault it is possible to recover from it and be ready for a new one in the future -- otherwise the syndrome register retains the old value.

Fixes https://github.com/ARMmbed/uvisor/issues/96
@meriac

AlessandroA commented:
@meriac updated after review

meriac commented on core/system/src/mpu/vmpu.c:
```diff
@@ -216,11 +218,18 @@ int vmpu_fault_recovery_bus(uint32_t pc, uint32_t sp)
+    /* check fault register; the following two configurations are allowed:
+     *   0x04 - imprecise data bus fault, no stacking/unstacking errors
+     *   0x82 - precise data bus fault, no stacking/unstacking errors */
+    if (SCB_BFSR == 0x82) {
```
As both branches of the if use SCB_BFSR : it might be more readable to use a switch statement ?

meriac commented on core/system/src/mpu/vmpu.c:
```diff
@@ -317,6 +326,9 @@ int vmpu_fault_recovery_bus(uint32_t pc, uint32_t sp)
     vmpu_unpriv_uint32_write(sp + (6 << 2), pc + ((UVISOR_NOP_CNT + 2 - cnt) << 1));
 
+    /* clear BFSR only - reset BFARVALID bit to acknowledge fault */
+    SCB_BFSR = 0x80;
```
SCB_BFSR needs to be cleared even in case the function returns - we might want to consider in this function to clear it right after checking it at the top.

AlessandroA commented:
@meriac Updated PR. Changelog now:
- moved check on fault registers right at the fault handler: create backup copy, clear flag, use backup copy in recovery functions
- removed checks from other functions (these resulted in removing one function, which is not needed anymore)
